### PR TITLE
Jetpack Section: Scan: Adds Scan History View

### PR DESCRIPTION
**Pull request by emilylaguna**
Project: #15190
Related PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/331
---

Adds scan history with error handling. 

The error messages aren't solidified yet and will need copy from @osullivanchris

#### General States
| Loading | All | Fixed | Ignored |
|:---:|:---:|:---:|:---:|
|loading|all|fixed|ignored|

#### Empty States
| No History | No Fixed | No Ignored | 
|:---:|:---:|:---:|
|nohistory|nofixed|noignored|

#### Error States
| Generic Error | No Internet |
|:---:|:---:|
|generic|nointernet|

### To test:
1. Launch the app, tap on the My Site, tap on a site with Jetpack Scan
2. Tap the history item in the top right corner
3. 👁️ You should see the scan history view with a loading indicator
4. 👁️ Once it finishes loading you'll see a list of threats
5. Tap the 'Fixed' tab item
6. 👁️ You'll only see a list of fixed threats (green color)
7. Tap the 'Ignored' tab item
8. 👁️ You'll only see a list of ignored threats (gray color)
9. Pull to refresh and wait for it to complete
8. 👁️ List is auto filtered using the tab item you had selected before

### PR submission checklist:

- [ ] I have considered adding unit tests where possible.
- [ ] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by momo-ozawa — COMMENTED** [review]
The site I'm testing has fixed threats but no ignored threats. It seems like section titles from the Fixed tab are still visible in the Ignored tab. If you scroll on the Ignored tab, you can see the cells from the Fixed tab scrolling as well.

Fixed | Ignored
--- | ---
Simulator Screen Shot - iPhone 12 Pro - 2021-01-25 at 10 56 06 | Simulator Screen Shot - iPhone 12 Pro - 2021-01-25 at 10 56 09

**Comment by jkmassel** [comment]
👋 Howdy! We are cutting the `16.6` release today.

Because of that, this PR will be bumped to `16.7`. If you need it to be part of `16.6`, please merge it into the `release/16.6` branch and DM me – I'll be happy to cut a new beta release!

**Comment by emilylaguna** [comment]
Ah good catch! Fixed in: 1cdbb5d6b49acd5d46e848f4c483362d5d0c3b5e

**Review by momo-ozawa — COMMENTED** [review]
Thanks for fixing the sections for empty views! Another thing I noticed about the empty views: when you swipe to refresh, the refresh control isn't displayed.

Otherwise, works as described 👍

File: WordPress/Classes/ViewRelated/Jetpack/Jetpack Scan/JetpackScanViewController.swift [hunk 1]
```diff
@@ -25,8 +25,15 @@ class JetpackScanViewController: UIViewController, JetpackScanView {
     override func viewDidLoad() {
         super.viewDidLoad()
 
+        self.title = NSLocalizedString("Scan", comment: "Title of the view")
+
         configureTableView()
         coordinator.viewDidLoad()
+
+        navigationItem.rightBarButtonItem = UIBarButtonItem(title: "History",
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Jetpack/Jetpack Scan/JetpackScanViewController.swift]: 👀 "History" button should be localized

Comment (emilylaguna) [reply WordPress/Classes/ViewRelated/Jetpack/Jetpack Scan/JetpackScanViewController.swift]: NooOooO!!!! Haha, done in e42ca2ca068b4651c256fbdf010b2d16fca062d3

**Review by momo-ozawa — APPROVED** [review]
> Another thing I noticed about the empty views: when you swipe to refresh, the refresh control isn't displayed.

I think this ^ can be addressed in a future PR, so I'm gonna go ahead and approve this PR 😄

**Pull request merged**